cirq/protocols/measurement_key_protocol.py

```diff
@@ -48,12 +56,21 @@ def _measurement_key_(self) -> str:
         will be stored.
         """
 
+    @document
+    def _measurement_keys_(self) -> Iterable[str]:
+        """Return the key that will be used to identify this measurement.
+
+        When a measurement occurs, either on hardware, or in a simulation,
+        this is the key value under which the results of the measurement
+        will be stored.
+        """
```

Comment:
One of the docstrings in this class should say whether it is ok to define both `_measurement_key_` and `_measurement_keys_` and what happens when one does this.

Reply:
Done.